src/StreamWrapper.php

```diff
@@ -106,6 +106,13 @@ public function stream_seek($offset, $whence)
         return true;
     }
 
+    public function stream_cast($cast_as)
+    {
+        $clonedStream = clone($this->stream);
```

Comment:
Why not just `$stream`?

Reply:
After call `detach` you will not be able to use `StreamWrapper` instance anymore because link to underlaying resource will be destroyed. For example:
https://github.com/guzzle/psr7/blob/master/src/Stream.php#L123
There is no another way to get underlaying stream.

Reply:
I mean: `$stream = clone($this->stream);`